File: dongyimai_user_web/src/main/webapp/js/controller/userController.js
 //用户表控制层 
app.controller('userController' ,function($scope,$controller,userService){
	
	$controller('baseController',{$scope:$scope});//继承
	
    //findAll
	$scope.findAll=function(){
		userService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//findPage
	$scope.findPage=function(page,rows){			
		userService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	
	//findOne
	$scope.findOne=function(id){				
		userService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	}
	
	//save（见于新增与修改功能共用一个保存按钮的情况）
	$scope.save=function(){				
		var serviceObject;//服务层对象  				
		if($scope.entity.id!=null){//如果有ID
			serviceObject=userService.update( $scope.entity ); //修改  
		}else{
			serviceObject=userService.add( $scope.entity  );//增加 
		}				
		serviceObject.success(
			function(response){
				if(response.success){
					//重新查询 
		        	$scope.reloadList();//重新加载
				}else{
					alert(response.message);
				}
			}		
		);				
	}

	//dele
	$scope.dele=function(){			
		//获取选中的复选框			
		userService.dele( $scope.selectIds ).success(
			function(response){
				if(response.success){
					$scope.reloadList();//刷新列表
					$scope.selectIds=[];
				}						
			}		
		);				
	}
	
	$scope.searchEntity={};//定义搜索对象
	//条件、模糊查询+分页
	$scope.search=function(page,rows){			
		userService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}

    /**
     * 生成验证码
     */
    $scope.createSmsCode = function (){
        if($scope.entity.phone==''||$scope.entity.phone==null){
            alert('手机号不能为空');
            return;
        }
        userService.createSmsCode($scope.entity.phone).success(
            function (response){
                if(response.success){
                    alert(response.message);
                }else{
                    alert(response.message);
                }
            })
    }

    /**
     *比对验证码，完成注册
     * @type {{}}
     */
    $scope.entity={};//初始化用户对象的数据结构
    //注册用户（用户名、密码、验证码判空，密码和确认密码判断一致）
    $scope.register = function (){
        if($scope.entity.username==''||$scope.entity.username==null){
            alert('用户名不能为空');
            return ;
        }
        if($scope.entity.password==''||$scope.entity.password==null){
            alert('密码不能为空');
            return ;
        }
        if($scope.entity.password!=$scope.repassword){
            alert('密码和确认密码需要一致！');
            return ;
        }
        if($scope.smsCode==''||$scope.smsCode==null){
            alert('验证码不能为空');
            return;
        }
        userService.add($scope.entity,$scope.smsCode).success(
            function (response){
                if(response.success){
                    alert('注册成功');
                }else{
                    alert(response.message);
                }
            })
    }


});	

File: dongyimai_user_web/src/main/java/com/offcn/user/controller/UserController.java
package com.offcn.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.offcn.entity.PageResult;
import com.offcn.entity.Result;
import com.offcn.pojo.TbUser;
import com.offcn.user.service.UserService;
import com.offcn.util.PhoneFormatCheckUtils;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/user")
public class UserController {

	@Reference
	private UserService userService;
	

	//dinfAll
	@RequestMapping("/findAll")
	public List<TbUser> findAll(){
		return userService.findAll();
	}
	
	//findPage
	@RequestMapping("/findPage")
	public PageResult findPage(int page, int rows){
		return userService.findPage(page, rows);
	}
	
	//add（已修改，需求：在新增用户之前，比对验证码）
	@RequestMapping("/add")
	public Result add(@RequestBody TbUser user, String smsCode){
		//比对验证码（缓存中的验证码与用户输入的验证码做比对）
		if(!userService.checkSmsCode(user.getPhone(),smsCode)){
			return new Result(false,"验证码错误！");
		}
		try {
			userService.add(user);
			return new Result(true, "增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "增加失败");
		}
	}
	
	//update
	@RequestMapping("/update")
	public Result update(@RequestBody TbUser user){
		try {
			userService.update(user);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}	
	
	//findOne
	@RequestMapping("/findOne")
	public TbUser findOne(Long id){
		return userService.findOne(id);		
	}
	
	//delete
	@RequestMapping("/delete")
	public Result delete(Long [] ids){
		try {
			userService.delete(ids);
			return new Result(true, "删除成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
	}
	
	//条件、模糊查询+分页
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbUser user, int page, int rows  ){
		return userService.findPage(user, page, rows);		
	}

	/**
	 * 生成验证码
	 * @param phone
	 * @return
	 */
	@RequestMapping("/createSmsCode")
	public Result createSmsCode(String phone){
		//判断手机号格式验证（前端验证码判空，后端验证判断格式）
		if(!PhoneFormatCheckUtils.isPhoneLegal(phone)){
			return new Result(false,"手机格式验证失败！");
		}
		try {
			userService.createSmsCode(phone);
			return new Result(true,"短信发送成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false,"短信发送失败");
		}
	}
	
}
